test/client/contract-client.js

```diff
@@ -396,14 +665,367 @@ describe('readonly calls', () => {
     })
   })
 
-  it('calls custom_readonly correctly', () => {
-    const nodeClient = sinon.fake()
+  it('calls readonly_bool correctly', () => {
+    const arg = true
+    const nodeClient = sinon.stub()
+    nodeClient.readonlySubmit = sinon.stub().withArgs(arg).returns(new Promise((resolve, reject) => {
```

Comment:
I think all of these inline `sinon.stub()` tests need to be rewritten. 
Check out: https://nikas.praninskas.com/javascript/2015/07/28/quickie-sinon-withargs-not-working/

I would recommend changing the stub definition as described in the article so it correctly only returns with the right params provided.

I reverted the change in the contract-client.js and all of these tests still passed, even though the params are not being provided correctly.  I added a console.log(params) to line 211 after readonly functions build the params which shows it only worked for string and custom types:

```
[]
    ✓ calls readonly_bool correctly
[]
    ✓ calls readonly_int32 correctly
[]
    ✓ calls readonly_uint32 correctly
[]
    ✓ calls readonly_f32 correctly
[]
    ✓ calls readonly_f64 correctly
[ 'abc' ]
    ✓ calls readonly_string correctly
[]
    ✓ calls readonly_bytes correctly
[ { one: '', two: 0, three: '' } ]
    ✓ calls readonly_custom correctly
```